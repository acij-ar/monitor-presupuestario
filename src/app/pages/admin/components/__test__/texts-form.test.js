const React = require('react');
const mockPost = jest.fn();
jest.mock('axios', () => ({ post: mockPost }));
const { shallow } = require('enzyme');

const TextsForm = require('../texts-form');

describe('Texts form', () => {
  const props = {
    texts: {
      "monitor": {
        "title": "Titulo Monitor",
        "description": "Descripción Monitor"
      },
      "about": {
        "rights": {
          "title": "Titulo derechos",
          "content": "Contenido derechos"
        },
        "methodology": {
          "title": "Titulo metodologia",
          "content": "Contenido metodologia"
        },
        "information": {
          "title": "Titulo información",
          "content": "Contenido informacion"
        },
        "dictionary": {
          "title": "Titulo Glosario",
          "content":"Contenido glosario"
        }
      }
    }
  };

  beforeEach(() => {
    mockPost.mockClear();
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<TextsForm {...props} />);
    expect(wrapper).toMatchSnapshot();

  });

  it('should post new text changes', async () => {
    mockPost.mockReturnValueOnce(Promise.resolve());
    const wrapper = shallow(<TextsForm {...props} />);
    wrapper.find('#monitor-title').simulate('change', { target: { value: 'new-monitor-title' } });
    wrapper.find('#monitor-description').simulate('change', { target: { value: 'new-monitor-description' } });
    wrapper.find('#rights-title').simulate('change', { target: { value: 'new-rights-title' } });
    wrapper.find('#rights-content').simulate('change', { target: { value: 'new-rights-description' } });
    wrapper.find('#methodology-title').simulate('change', { target: { value: 'new-methodology-title' } });
    wrapper.find('#methodology-content').simulate('change', { target: { value: 'new-methodology-description' } });
    wrapper.find('#information-title').simulate('change', { target: { value: 'new-information-title' } });
    wrapper.find('#information-content').simulate('change', { target: { value: 'new-information-description' } });
    wrapper.find('#dictionary-title').simulate('change', { target: { value: 'new-dictionary-title' } });
    wrapper.find('#dictionary-content').simulate('change', { target: { value: 'new-dictionary-description' } });

    await wrapper.find('button').simulate('click');
    const expectedTexts = {
      monitor: {
        title: 'new-monitor-title',
        description: 'new-monitor-description',
      },
      about: {
        rights: {
          title: 'new-rights-title',
          content: 'new-rights-description',
        },
        methodology: {
          title: 'new-methodology-title',
          content: 'new-methodology-description',
        },
        information: {
          title: 'new-information-title',
          content: 'new-information-description',
        },
        dictionary: {
          title: 'new-dictionary-title',
          content: 'new-dictionary-description',
        },
      },
    };
    expect(mockPost).toHaveBeenCalledWith('/api/admin/texts', { texts: expectedTexts });
  });

  it('should post new text changes and show possitive feedback when success', async () => {
    mockPost.mockReturnValueOnce(Promise.resolve());
    const wrapper = shallow(<TextsForm {...props} />);
    await wrapper.find('button').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('should post new text changes and show negative feedback when fails', async () => {
    mockPost.mockReturnValueOnce(Promise.reject());
    const wrapper = shallow(<TextsForm {...props} />);
    await wrapper.find('button').simulate('click');
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});