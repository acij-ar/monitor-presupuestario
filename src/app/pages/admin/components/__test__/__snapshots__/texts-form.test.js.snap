// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Texts form should match snapshot 1`] = `
<div
  className="monitor-content monitor-admin"
>
  <div
    className="monitor-admin-page-section"
  >
    <h2>
      Monitor
    </h2>
    <h3>
      Título
    </h3>
    <input
      id="monitor-title"
      onChange={[Function]}
      value="Titulo Monitor"
    />
    <h3>
      Descripción
    </h3>
    <textarea
      id="monitor-description"
      onChange={[Function]}
      value="Descripción Monitor"
    />
  </div>
  <div
    className="monitor-admin-page-section"
  >
    <h2>
      Acerca de
    </h2>
    <h3>
      Presupuesto para los derechos
    </h3>
    <input
      id="rights-title"
      onChange={[Function]}
      value="Titulo derechos"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="rights-content"
      onChange={[Function]}
      value="Contenido derechos"
    />
    <h3>
      Metodologia
    </h3>
    <input
      id="methodology-title"
      onChange={[Function]}
      value="Titulo metodologia"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="methodology-content"
      onChange={[Function]}
      value="Contenido metodologia"
    />
    <h3>
      Publicación de la información
    </h3>
    <input
      id="information-title"
      onChange={[Function]}
      value="Titulo información"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="information-content"
      onChange={[Function]}
      value="Contenido informacion"
    />
    <h3>
      Glosario
    </h3>
    <input
      id="dictionary-title"
      onChange={[Function]}
      value="Titulo Glosario"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="dictionary-content"
      onChange={[Function]}
      value="Contenido glosario"
    />
  </div>
  <button
    onClick={[Function]}
  >
    Guardar
  </button>
</div>
`;

exports[`Texts form should post new text changes and show negative feedback when fails 1`] = `
<div
  className="monitor-content monitor-admin"
>
  <div
    className="monitor-admin-page-section"
  >
    <h2>
      Monitor
    </h2>
    <h3>
      Título
    </h3>
    <input
      id="monitor-title"
      onChange={[Function]}
      value="new-monitor-title"
    />
    <h3>
      Descripción
    </h3>
    <textarea
      id="monitor-description"
      onChange={[Function]}
      value="new-monitor-description"
    />
  </div>
  <div
    className="monitor-admin-page-section"
  >
    <h2>
      Acerca de
    </h2>
    <h3>
      Presupuesto para los derechos
    </h3>
    <input
      id="rights-title"
      onChange={[Function]}
      value="new-rights-title"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="rights-content"
      onChange={[Function]}
      value="new-rights-description"
    />
    <h3>
      Metodologia
    </h3>
    <input
      id="methodology-title"
      onChange={[Function]}
      value="new-methodology-title"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="methodology-content"
      onChange={[Function]}
      value="new-methodology-description"
    />
    <h3>
      Publicación de la información
    </h3>
    <input
      id="information-title"
      onChange={[Function]}
      value="new-information-title"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="information-content"
      onChange={[Function]}
      value="new-information-description"
    />
    <h3>
      Glosario
    </h3>
    <input
      id="dictionary-title"
      onChange={[Function]}
      value="new-dictionary-title"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="dictionary-content"
      onChange={[Function]}
      value="new-dictionary-description"
    />
  </div>
  <button
    onClick={[Function]}
  >
    Guardar
  </button>
  <span>
    ❌
  </span>
</div>
`;

exports[`Texts form should post new text changes and show possitive feedback when success 1`] = `
<div
  className="monitor-content monitor-admin"
>
  <div
    className="monitor-admin-page-section"
  >
    <h2>
      Monitor
    </h2>
    <h3>
      Título
    </h3>
    <input
      id="monitor-title"
      onChange={[Function]}
      value="new-monitor-title"
    />
    <h3>
      Descripción
    </h3>
    <textarea
      id="monitor-description"
      onChange={[Function]}
      value="new-monitor-description"
    />
  </div>
  <div
    className="monitor-admin-page-section"
  >
    <h2>
      Acerca de
    </h2>
    <h3>
      Presupuesto para los derechos
    </h3>
    <input
      id="rights-title"
      onChange={[Function]}
      value="new-rights-title"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="rights-content"
      onChange={[Function]}
      value="new-rights-description"
    />
    <h3>
      Metodologia
    </h3>
    <input
      id="methodology-title"
      onChange={[Function]}
      value="new-methodology-title"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="methodology-content"
      onChange={[Function]}
      value="new-methodology-description"
    />
    <h3>
      Publicación de la información
    </h3>
    <input
      id="information-title"
      onChange={[Function]}
      value="new-information-title"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="information-content"
      onChange={[Function]}
      value="new-information-description"
    />
    <h3>
      Glosario
    </h3>
    <input
      id="dictionary-title"
      onChange={[Function]}
      value="new-dictionary-title"
    />
    <h3>
      Contenido
    </h3>
    <textarea
      id="dictionary-content"
      onChange={[Function]}
      value="new-dictionary-description"
    />
  </div>
  <button
    onClick={[Function]}
  >
    Guardar
  </button>
  <span>
    ✅
  </span>
</div>
`;
